### PR TITLE
Region Equality and Rotation improvements

### PR DESCRIPTION
**Pull request by fcheadle**
removed equality / get hash code, implemented IMatches<Region>, rotate now uses the primitives rotation and deleted the method from region.

If accepted, this will close issues numbers #183 and #184

### CODE REVIEW

File: GoRogue/MapGeneration/Region.cs [hunk 1]
```diff
@@ -42,36 +42,30 @@ public Region(string name, Point northWest, Point northEast, Point southEast, Po
...
+
         /// <summary>
-        /// Returns a string detailing the region's corner locations.
+        /// Returns whether or not this region is completely overlapping with the other region
         /// </summary>
-        /// <returns/>
-        public override string ToString()
+        /// <param name="other">the region against which to compare</param>
+        /// <returns></returns>
+        public bool Matches(Region other)
```
Comment (Chris3606) [inline GoRogue/MapGeneration/Region.cs:51]: Matching requires the input type to be nullable: this needs to be:
```suggestion
        public bool Matches(Region? other)
```

And needs to account for the input parameter possibly being null.  This should get rid of the requisite warnings during CI as well.

File: GoRogue/MapGeneration/Region.cs [hunk 2]
```diff
@@ -332,15 +325,15 @@ public virtual Region Rotate(double degrees, Point origin)
         /// Adds a sub-region to this region.
         /// </summary>
         /// <param name="region">The region you wish to add as a sub-region</param>
-        public void Add(Region region)
+        public void AddRegion(Region region)
```
Comment (Chris3606) [inline GoRogue/MapGeneration/Region.cs:328]: Why `AddRegion` instead of `AddSubRegion?`

File: GoRogue/MapGeneration/Region.cs [hunk 3]
```diff
@@ -332,15 +325,15 @@ public virtual Region Rotate(double degrees, Point origin)
...
         {
             _subRegions.Add(region);
         }
         /// <summary>
         /// Removes a sub-region whose name matches with the string given.
         /// </summary>
         /// <param name="name">Key of sub-region to remove.</param>
-        public void Remove(string name)
+        public void RemoveRegion(string name)
```
Comment (Chris3606) [inline GoRogue/MapGeneration/Region.cs:336]: See above; would `RemoveSubRegion` be better?

File: GoRogue/GoRogue.csproj [hunk 4]
```diff
@@ -84,7 +84,7 @@
     </PackageReference>
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
     <PackageReference Include="OptimizedPriorityQueue" Version="4.2.0" />
-    <PackageReference Include="TheSadRogue.Primitives" Version="1.0.0-alpha5" />
+    <PackageReference Include="TheSadRogue.Primitives" Version="1.0.0-alpha6-debug" />
```
Comment (Chris3606) [inline GoRogue/GoRogue.csproj:87]: Shouldn't default to debug package here; should be:
```suggestion
    <PackageReference Include="TheSadRogue.Primitives" Version="1.0.0-alpha6" />
```

**Review by Chris3606 — CHANGES_REQUESTED** [review]
One nullability issue that we need to fix (see comments and GitHub actions results for details).

Beyond that, just some naming items that might improve readability and a minor package reference switch.  These changes will conflict with the other PR I have in, but the other PR's changes to region are pretty much equivalent so I should be able to fix that on my end.

File: GoRogue/MapGeneration/Region.cs (hunk 2 above)
Comment (fcheadle) [reply GoRogue/MapGeneration/Region.cs:328]: No particular reason. Changed in new version

File: GoRogue/MapGeneration/Region.cs (hunk 3 above)
Comment (fcheadle) [reply GoRogue/MapGeneration/Region.cs:336]: Agreed

File: GoRogue/GoRogue.csproj (hunk 4 above)
Comment (fcheadle) [reply GoRogue/GoRogue.csproj:87]: Good catch. Thanks

File: GoRogue/MapGeneration/Region.cs (hunk 1 above)
Comment (fcheadle) [reply GoRogue/MapGeneration/Region.cs]: Excellent catch. Fixed in new version

**Review by Chris3606 — APPROVED** [review]
Looks good, merging shortly.

**Pull request merged**